### PR TITLE
Feature/cb 71, feature/modals

### PR DESCRIPTION
**Pull request by DenilFuirie**

### CODE REVIEW

File: src/pages/MyContracts/MyContracts.tsx [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+          md={12}
+          lg={12}
+          xl={12}
+          key="123"
+        >
+          <TextField
+            id="input-with-icon-textfield"
+            placeholder="Search contract"
+            onChange={(e) => searchHandler(e.target.value)}
+            InputProps={{
```
Comment (ValnetineKul) [inline src/pages/MyContracts/MyContracts.tsx:68]:
```
InputProps={{
              startAdornment: <SearchIcon />,
            }}
```

File: src/pages/MyContracts/MyContracts.tsx [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+      } return true;
+    });
+    setFilteredCards([...newState]);
+  }, [searchValue, debouncedSearchValue]);
+
+  const searchHandler = useCallback((value) => {
+    setSearchValue(value);
+  }, [debouncedSearchValue]);
+
+  useEffect(() => {
```
Comment (ValnetineKul) [inline src/pages/MyContracts/MyContracts.tsx:48]: why do you need useEffect to watch for changes if you cant filter in searchHandler directly?

File: src/pages/MyContracts/MyContracts.tsx [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+        {filteredCards.map(({
+          contractName,
+          contractDate,
+          contractType,
+          contractLogo,
+          contractButtons,
+          isRequestBlockActive,
+          contractKey,
+        }) => (
+          <Grid
```
Comment (ValnetineKul) [inline src/pages/MyContracts/MyContracts.tsx:86]: grids with 12 columns in each breakpoint make no sense, use box

File: src/pages/MyContracts/MyContracts.styles.tsx [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+    padding: `${theme.spacing(3)}px`,
+    width: '100%',
+    '& > :nth-child(1)': {
+      marginRight: theme.spacing(2),
+    },
+  },
+  contractBottom: {
+    ...flexHelper('space-between'),
+    padding: theme.spacing(3),
+    flexWrap: 'nowrap',
```
Comment (ValnetineKul) [inline src/pages/MyContracts/MyContracts.styles.tsx:42]: this is default value of flexwrap, dont write defautl values

File: src/components/CompleteModal/CompleteModal/CompleteModal.tsx [hunk 5]
```diff
@@ -0,0 +1,51 @@
+import React, { VFC } from 'react';
+
+import clsx from 'clsx';
+import { Modal } from 'components/Modal';
+import { Box, Typography } from '@material-ui/core';
+import { useStyles } from './CompleteModal.styles';
+import { SuccessBigIcon } from '../../../theme/icons/components/SuccessBigIcon';
```
[the remark this answers is not in the document]
Comment (DenilFuirie) [reply src/components/CompleteModal/CompleteModal/CompleteModal.tsx:7]: fixed

File: src/components/CompleteModal/CompleteModal/CompleteModal.tsx [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+  result?: boolean;
+  isLoading?: boolean;
+}
+
+export const CompleteModal: VFC<PaymentModalProps> = ({
+  open, onClose, result,
+}) => {
+  const classes = useStyles();
+
+  const showSuccess = (
```
[the remark this answers is not in the document]
Comment (DenilFuirie) [reply src/components/CompleteModal/CompleteModal/CompleteModal.tsx:25]: fixed

File: src/components/CompleteModal/CompleteModal/CompleteModal.tsx [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+
+  const showError = (
+    <>
+      <ErrorBigIcon />
+      <Typography className={classes.desc} variant="body2"> Transaction error !</Typography>
+      <Typography className={classes.desc} variant="body2"> Please  come back later</Typography>
+    </>
+  );
+
+  const showResult = (
```
[the remark this answers is not in the document]
Comment (DenilFuirie) [reply src/components/CompleteModal/CompleteModal/CompleteModal.tsx:40]: fixed

File: src/components/CompleteModal/CompleteModal/CompleteModal.styles.tsx [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+  return createStyles({
+    root: {
+      [formatMedia.BREAKPOINT_TABLET]: {
+      },
+    },
+    icon: {
+      ...flexHelper(),
+      flexDirection: 'column',
+    },
+    desc: {
```
[the remark this answers is not in the document]
Comment (DenilFuirie) [reply src/components/CompleteModal/CompleteModal/CompleteModal.styles.tsx:19]: fixed

File: src/components/Loader/Loader.styles.tsx [hunk 9]
```diff
@@ -0,0 +1,57 @@
+/* eslint-disable */
+import { makeStyles } from '@material-ui/core';
+
+export const useStyles = makeStyles({
+  loaderModal: {
+    background: 'transparent',
+  },
+  root: {
+    position: 'fixed',
+    background: 'rgba(0, 0, 0, .5)',
```
[the remark this answers is not in the document]
Comment (DenilFuirie) [reply src/components/Loader/Loader.styles.tsx:10]: fixed

File: src/components/Loader/Loader.styles.tsx [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+    top: 0,
+    left: 0,
+    zIndex: 9999999999999999999,
+    display: 'inline-block',
```
[the remark this answers is not in the document]
Comment (DenilFuirie) [reply src/components/Loader/Loader.styles.tsx:14]: fixed

File: src/components/Loader/Loader.styles.tsx [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+    width: '100%',
+    height: '100%',
+  },
+
+  loader: {
+    zIndex: 9999999999999999999,
+    width: '100%',
+    height: '100%',
+    backgroundColor: 'transparent',
+    display: 'flex',
```
[the remark this answers is not in the document]
Comment (DenilFuirie) [reply src/components/Loader/Loader.styles.tsx:26]: fixed

File: src/components/Loader/Loader.styles.tsx [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+    to: {
+      transform: 'rotate(360deg)',
+    },
+  },
+
+  icon: {
+    width: '100px',
+    height: 'auto',
+    zIndex: 9999999999999999999,
+    // animation: 'imageFloating 1s linear infinite',
```
[the remark this answers is not in the document]
Comment (DenilFuirie) [reply src/components/Loader/Loader.styles.tsx:44]: fixed

File: src/components/Preview/Preview.tsx [hunk 13]
```diff
@@ -12,6 +13,9 @@ import { useNavigate } from 'react-router-dom';
 import { routes } from 'appConstants';
 import { useStyles } from './Preview.styles';
 import { iconHelper } from './Preview.helpers';
+import { CompleteModal } from '../CompleteModal/CompleteModal';
+import { Loader } from '../Loader';
```
[the remark this answers is not in the document]
Comment (DenilFuirie) [reply src/components/Preview/Preview.tsx:17]: fixed

File: src/components/Preview/Preview.tsx [hunk 14]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
```
[the remark this answers is not in the document]
Comment (DenilFuirie) [reply src/components/Preview/Preview.tsx:1]: fixed

File: src/components/Preview/Preview.tsx [hunk 15]
```diff
@@ -55,10 +62,23 @@ export const Preview: FC<PreviewProps> = ({
...
+    setCompleteOpen(true);
+  }, []);
+
+  const closeCompleteModal = useCallback(() => {
+    setCompleteOpen(false);
+  }, []);
+
   const onPay = useCallback(() => {
-    launchAction();
+    // launchAction();
```
[the remark this answers is not in the document]
Comment (DenilFuirie) [reply src/components/Preview/Preview.tsx:74]: fixed

File: src/components/Preview/Preview.tsx [hunk 16]
```diff
@@ -55,10 +62,23 @@ export const Preview: FC<PreviewProps> = ({
...
     closePaymentModal();
-    navigate(routes.root);
+    setIsLoading(true);
+    setTimeout(() => {
+      setIsLoading(false);
+      openCompleteModal();
+    }, 1000);
+    // navigate(routes.root);
```
[the remark this answers is not in the document]
Comment (DenilFuirie) [reply src/components/Preview/Preview.tsx:81]: fixed, return back navigate

File: src/hooks/useDebounce.ts [hunk 17]
```diff
@@ -0,0 +1,20 @@
+import { useEffect, useState } from 'react';
```
[the remark this answers is not in the document]
Comment (DenilFuirie) [reply src/hooks/useDebounce.ts:1]: fixed

File: src/pages/MyContracts/MyContracts.helpers.tsx [hunk 18]
```diff
@@ -0,0 +1,97 @@
+import React from 'react';
+import {
+  ContractToken, CrowdsaleIcon, WeddingRingIcon, WillContract,
+} from '../../theme/icons';
```
[the remark this answers is not in the document]
Comment (DenilFuirie) [reply src/pages/MyContracts/MyContracts.helpers.tsx:4]: fixed

File: src/pages/MyContracts/MyContracts.helpers.tsx [hunk 19]
```diff
@@ -0,0 +1,97 @@
...
+    contractButtons: [
+      {
+        type: 'viewContract',
+        title: 'View contract',
+      },
+    ],
+  },
+  {
+    contractKey: '1',
+    contractDate: '22.01.2022',
```
[the remark this answers is not in the document]
Comment (DenilFuirie) [reply src/pages/MyContracts/MyContracts.helpers.tsx:41]: fixed

File: src/pages/MyContracts/MyContracts.styles.tsx [hunk 20]
```diff
@@ -0,0 +1,105 @@
...
+export const useStyles = makeStyles((theme: Theme) => createStyles({
+  root: {
+    width: '90%',
+    [theme.breakpoints.down('sm')]: {
+      width: '100%',
+    },
+  },
+  contractBlock: {
+    ...flexHelper(),
+    padding: theme.spacing(0),
```
[the remark this answers is not in the document]
Comment (DenilFuirie) [reply src/pages/MyContracts/MyContracts.styles.tsx:20]: fixed

File: src/pages/MyContracts/MyContracts.styles.tsx [hunk 21]
```diff
@@ -0,0 +1,105 @@
...
+    background: theme.palette.type === 'dark' ? COLOR_BLACK_1 : COLOR_BUTTON_PRIMARY_LIGHT_DEFAULT,
+  },
+  contractHead: {
+    ...flexHelper('space-between'),
+    width: '100%',
+    padding: `${theme.spacing(4)}px ${theme.spacing(4)}px 0px ${theme.spacing(4)}px`,
+  },
+  contractTitle: {
+    ...flexHelper('flex-start'),
+    padding: `${theme.spacing(3)}px`,
```
[the remark this answers is not in the document]
Comment (DenilFuirie) [reply src/pages/MyContracts/MyContracts.styles.tsx:33]: fixed

File: src/pages/MyContracts/MyContracts.styles.tsx [hunk 22]
```diff
@@ -0,0 +1,105 @@
...
+    },
+  },
+  contractActionBlock: {
+    ...flexHelper('space-between'),
+    padding: `${theme.spacing(1.5)}px ${theme.spacing(3)}px`,
+    width: '100%',
+    background: theme.palette.type === 'dark' ? COLOR_BLACK_3 : COLOR_GREY_4,
+    [theme.breakpoints.down('sm')]: {
+      flexDirection: 'column',
+      '& > p': {
```
[the remark this answers is not in the document]
Comment (DenilFuirie) [reply src/pages/MyContracts/MyContracts.styles.tsx:62]: fixed

File: src/pages/MyContracts/MyContracts.tsx [hunk 23]
```diff
@@ -0,0 +1,137 @@
...
+    <Container>
+      <Grid container className={classes.root}>
+        <Grid
+          item
+          xs={12}
+          sm={12}
+          md={12}
+          lg={12}
+          xl={12}
+          key="123"
```
[the remark this answers is not in the document]
Comment (DenilFuirie) [reply src/pages/MyContracts/MyContracts.tsx:62]: fixed

File: src/pages/MyContracts/MyContracts.tsx [hunk 24]
```diff
@@ -0,0 +1,137 @@
...
+  }, [debouncedSearchValue]);
+
+  useEffect(() => {
+    filterData();
+  }, [debouncedSearchValue]);
+
+  return (
+    <Container>
+      <Grid container className={classes.root}>
+        <Grid
```
[the remark this answers is not in the document]
Comment (DenilFuirie) [reply src/pages/MyContracts/MyContracts.tsx:55]: fixed

File: src/pages/MyContracts/MyContracts.tsx [hunk 25]
```diff
@@ -0,0 +1,137 @@
+/* eslint-disable react/no-array-index-key,no-param-reassign */
+import React, { useCallback, useEffect, useState } from 'react';
+import {
+  Box, Button,
+  Container, Grid, IconButton, InputAdornment, TextField, Typography,
+} from '@material-ui/core';
+import clsx from 'clsx';
+import { useStyles } from './MyContracts.styles';
+import { SearchIcon } from '../../theme/icons/components/SearchIcon';
```
[the remark this answers is not in the document]
Comment (DenilFuirie) [reply src/pages/MyContracts/MyContracts.tsx:9]: fixed

File: src/pages/MyContracts/MyContracts.tsx [hunk 26]
```diff
@@ -0,0 +1,137 @@
...
+                    onClick={() => buttonClickHandler(contractKey, type)}
+                    className={classes.button}
+                    value={type}
+                    key={`${type}_${index}`}
+                    variant="outlined"
+                  >{title}
+                  </Button>
+                ))}
+              </Box>
+              <Grid item className={classes.chainTagContainer}>
```
[the remark this answers is not in the document]
Comment (DenilFuirie) [reply src/pages/MyContracts/MyContracts.tsx:128]: fixed

File: src/components/Loader/Loader.styles.tsx [hunk 27]
```diff
@@ -0,0 +1,57 @@
...
+export const useStyles = makeStyles({
+  loaderModal: {
+    background: 'transparent',
+  },
+  root: {
+    position: 'fixed',
+    background: 'rgba(0, 0, 0, .5)',
+    top: 0,
+    left: 0,
+    zIndex: 9999999999999999999,
```
[the remark this answers is not in the document]
Comment (DenilFuirie) [reply src/components/Loader/Loader.styles.tsx:13]: fixed

File: src/pages/MyContracts/MyContracts.styles.tsx [hunk 28]
```diff
@@ -0,0 +1,105 @@
...
+      ...flexHelper('flex-start'),
+    },
+  },
+  button: {
+    height: 56,
+    fontSize: 18,
+    minWidth: 141,
+    background: theme.palette.type === 'dark' ? COLOR_BLACK_1 : COLOR_BUTTON_SECONDARY_LIGHT_DEFAULT,
+    textTransform: 'none',
+    '&:not(:last-child)': {
```
[the remark this answers is not in the document]
Comment (DenilFuirie) [reply src/pages/MyContracts/MyContracts.styles.tsx:91]: fixed

File: src/pages/MyContracts/MyContracts.tsx (hunk 1 above)
Comment (DenilFuirie) [reply src/pages/MyContracts/MyContracts.tsx:68]: fixed

File: src/pages/MyContracts/MyContracts.tsx (hunk 2 above)
Comment (DenilFuirie) [reply src/pages/MyContracts/MyContracts.tsx:48]: fixed

File: src/pages/MyContracts/MyContracts.tsx (hunk 3 above)
Comment (DenilFuirie) [reply src/pages/MyContracts/MyContracts.tsx:86]: fixed

File: src/pages/MyContracts/MyContracts.styles.tsx (hunk 4 above)
Comment (DenilFuirie) [reply src/pages/MyContracts/MyContracts.styles.tsx:42]: fixed

File: src/pages/MyContracts/MyContracts.tsx [hunk 29]
```diff
@@ -0,0 +1,137 @@
...
+import { NetTag } from '../../containers/Header/components/NetTag';
+import { contractsCards } from './MyContracts.helpers';
+import useDebounce from '../../hooks/useDebounce';
+
+export const MyContracts = () => {
+  const [cards, setCards] = useState(contractsCards);
+  const [filteredCards, setFilteredCards] = useState(contractsCards);
+  const [searchValue, setSearchValue] = useState<string>('');
+  const classes = useStyles();
+  const isMainnet = true;
```
[the remark this answers is not in the document]
Comment (DenilFuirie) [reply src/pages/MyContracts/MyContracts.tsx]: fixed

**Pull request merged**